### PR TITLE
Add systemd service to APT package HZ-673

### PR DESCRIPTION
**Pull request by ldziedziul**
- Adds systemd service script to deb package
- During installation adds `hazelcast` user for running the service 

APT counterpart of https://github.com/hazelcast/hazelcast-packaging/pull/120

## Testing 
- on ubuntu docker built the deb package:
```shell
HZ_DISTRIBUTION=hazelcast HZ_VERSION=5.1-SNAPSHOT PACKAGE_VERSION=5.1-SNAPSHOT ./build-hazelcast-deb-package.sh
```
- on a full-blown Ubuntu VM (docker Ubuntu doesn't have systemd):
- installed deb file `dpkg -i ./hazelcast-5.1-SNAPSHOT-all.deb` 
- tested if the following commands work: 
   - systemctl status hazelcast
   - systemctl start hazelcast
   - systemctl status hazelcast
   - systemctl stop hazelcast
   - systemctl enable hazelcast (and verify after reboot that service is up and running)
   - systemctl disable hazelcast
- tested removal of running service
   - systemctl enable hazelcast
   - systemctl start hazelcast
   - apt remove hazelcast
   - systemctl status hazelcast (verify service not found)

### CODE REVIEW

File: packages/deb/hazelcast/DEBIAN/hazelcast.service [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[Unit]
```
Comment (ldziedziul) [inline packages/deb/hazelcast/DEBIAN/hazelcast.service:1]: @frant-hartm we should unify and use a single service file for both packages

**Review by frant-hartm — CHANGES_REQUESTED** [review]

File: packages/deb/hazelcast/DEBIAN/hazelcast.service [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+[Service]
+User=hazelcast
+Group=hazelcast
+
+SuccessExitStatus=143
+
+Type=simple
+
+ExecStart=hz start
```
[the remark this answers is not in the document]
Comment (ldziedziul) [reply packages/deb/hazelcast/DEBIAN/hazelcast.service]: Fixed: https://github.com/hazelcast/hazelcast-packaging/pull/121/commits/04e1dfe5276ec6e8db45dac9d351f9db3caf7d64

**Comment by ldziedziul** [comment]
@frant-hartm I've moved `hazelcast.service` to `packages/common/hazelcast.service`

File: build-hazelcast-deb-package.sh [hunk 3]
```diff
@@ -42,14 +43,13 @@ if [ "${HZ_DISTRIBUTION}" == "hazelcast-enterprise" ]; then
...
-envsubst <packages/deb/hazelcast/DEBIAN/conffiles >build/deb/hazelcast/DEBIAN/conffiles
+# Replace variable placeholders in the following files with the values from the environment
 envsubst <packages/deb/hazelcast/DEBIAN/control >build/deb/hazelcast/DEBIAN/control
-envsubst <packages/deb/hazelcast/DEBIAN/postinst >build/deb/hazelcast/DEBIAN/postinst
-envsubst <packages/deb/hazelcast/DEBIAN/prerm >build/deb/hazelcast/DEBIAN/prerm
+
+cp packages/deb/hazelcast/DEBIAN/conffiles build/deb/hazelcast/DEBIAN/conffiles
+cp packages/deb/hazelcast/DEBIAN/postinst build/deb/hazelcast/DEBIAN/postinst
+cp packages/deb/hazelcast/DEBIAN/prerm build/deb/hazelcast/DEBIAN/prerm
+cp packages/common/hazelcast.service build/deb/hazelcast/lib/systemd/system/hazelcast.service
```
Comment (frant-hartm) [inline build-hazelcast-deb-package.sh:52]: Nice cleanup here :+1:

**Review by frant-hartm — APPROVED** [review]

**Pull request merged**